src/idpf.rs

```diff
@@ -135,22 +136,70 @@ where
 /// Trait for values to be programmed into an IDPF.
 ///
 /// Values must form an Abelian group, so that they can be secret-shared, and the group operation
-/// must be represented by [`Add`]. An implementation of [`CoinToss`] must be provided to randomly
-/// select a value using PRG output. Values must be encodable and decodable, without need for a
-/// decoding parameter.
+/// must be represented by [`Add`]. Values must be encodable and decodable, without need for a
+/// decoding parameter. Values can be pseudorandomly generated, with a uniform probability
+/// distribution, from PRG output.
 pub trait IdpfValue:
     Add<Output = Self>
     + AddAssign
     + Sub<Output = Self>
-    + ConditionallySelectable
     + ConditionallyNegatable
-    + CoinToss
     + Encode
     + Decode
     + Sized
 {
+    /// Any run-time parameters needed to produce a value.
+    type Parameters;
+
+    /// Generate a value from PRG output. This must have a uniform probability distribution over
+    /// the PRG output.
+    fn generate<S>(seed_stream: &mut S, parameters: &Self::Parameters) -> Self
+    where
+        S: SeedStream;
+
     /// Returns the additive identity.
-    fn zero() -> Self;
+    fn zero(parameters: &Self::Parameters) -> Self;
+
+    /// Conditionally select between two values in constant time.
```

Comment:
Make it clear that this is not a property of any implementation, but a design goal.
```suggestion
    /// Conditionally select between two values. Implementations must perform this operation in constant time.
```